utils/data_processing.py

```diff
@@ -0,0 +1,44 @@
+"""Process and format the mesh data for our model"""
+
+from datatypes.graph import EdgeSet
+from datatypes.graph import Graph
+
+import numpy as np
+
+
+def _triangles_to_edges(faces):
+
+    edges = np.concatenate([faces[:, 0:2], faces[:, 1:3], faces[:, [0, 2]]],
+                           axis=0)
+
+    edges = np.sort(edges, axis=1)
+    unique_edges = np.unique(edges, axis=0)
+
+    senders, receivers = unique_edges[:, 0], unique_edges[:, 1]
+
+    return (np.concatenate([senders,
```

Comment:
Ok, I see the point here. Does this structure arise from the literature you have seen?

I see the point, I just want to be sure if it is truly necessary...

Reply:
Yes, this follows MeshGraphNets implementation. Edges are present twice, once in each direction. I believe it is coherent with the message passing implementation in their model but I will make sure later if this is indeed necessary

Reply:
Alright! It was just to be clear.